Consolidate properties at project level

Re4son opened the pull request:
Define common properties at project level:
    compileSdkVersion=28
    minSdkVersion=14
    targetSdkVersion=28
    versionName="2019.2-rc1"

Downgrade gradle version requirements to 3.2

The above fixes the problem that the app only runs on >=Oreo. One of the modules gradle,build had a wrong minSdkVersion. Shouldn't happen in the future anymore.
The app seems to work well but it terminates not in the most graceful manner, i.e. it throws a "NetHunter Terminal has stopped" error. Is it working for you?

Re4son committed:
Consolidate properties at project level

Define common properties at project level:
    compileSdkVersion=28
    minSdkVersion=14
    targetSdkVersion=28
    versionName="2019.2-rc1"

Downgrade gradle version requirements to 3.2

simonpunk commented:
@Re4son 
Hi, thanks for submitting the pull request, i am really appreciated.

Actually I still don't know what gradle is all about but the target and min sdk version, and I forgot to change the minSdk back to 14 in one of the gradle as well.

Also, you mentioned your own built nh_term throws an error, but for me I didn't encounter any issue like this with my own built, did you install it on lower android version like nougat or lolipop? As I haven't test on any devices with lower android verison, maybe take a look at what the logcat says. However, what about installing my built apk, does it still throw the same error as well?

Is the gradle version really needs to be 3.2? I have changed all the minSdk to 14 and target sdk to 28, and currently I am using "com.android.tools.build:gradle:3.4.0-beta05" and it still can build without issues and the app is working as well. Many thanks.

Re4son commented:
I haven't debugged it yet but it seems to run nicely on my Oreo phone but throws an error on Nougat.

I reduced the required version of Android Studio from "3.4.0-alpha09" to "3.2" to ensure the project can be opened with Android Studio on a current version -1 in the stable channel and everything newer.

Re4son committed:
Restart NotificationManager on exit if required

Closing the application on Android Nougat and older causes a null
pointer exception so we check if the NotificatiionManager Service
exists and start it if it doesn't

Re4son committed:
Add versionCode and bump version to 2019.2-rc2

Re4son commented:
Fixed it.
On another note: Do you mind if we move my fork into the offensive-security repo and push future changes into that? We've got too many different versions already and that would make it easier to keep track of it.

Re4son committed:
Update README.md

simonpunk commented:
> Fixed it.
> On another note: Do you mind if we move my fork into the offensive-security repo and push future changes into that? We've got too many different versions already and that would make it easier to keep track of it.

Absolutely the best way, pease do it . Because i thought the official repo is no longer maintained, so I just forked it and tried to fix it myself. But one more issue is that the hostname is not dispalyed in Android and android su terminal probably on Pie or Oreo, I havent looked into it, it would be great if you can fix it as well. Later I will merge all the pull requests, thank you so much and let me learn new things.
